web/src/echo_state.ts

```diff
@@ -0,0 +1,33 @@
+import type {Message} from "./message_store";
+
+const waiting_for_id = new Map<number, Message>();
+let waiting_for_ack = new Map<number, Message>();
```

Comment:
In `echo.insert_local_message`, `message.local_id` is a `string`, which is the key in maps `waiting_for_ack` and `waiting_for_id`. Hence 
```typescript
const waiting_for_id = new Map<string, Message>();
let waiting_for_ack = new Map<string, Message>();
```

Reply:
oh ya, right, updated!